modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/metadata/MetadataMediator.java

```diff
@@ -171,7 +171,7 @@ public MetadataMediator(T componentModel) {
       }
 
       keyValue = keyValueResult.get();
-      if (keyValue == null && !keyIdObjectResolver.isKeyLess()) {
+      if (keyValue == null && !keyIdObjectResolver.isKeyRequired()) {
```

Comment:
I think that this should fail if key is required. It looks like this will fail when key is not required.